kyber_group.go

```diff
@@ -92,18 +92,29 @@ func NewGroupGT() kyber.Group {
 	}
 }
 
-type Suite struct{}
+type Suite struct {
+	domainG1 []byte
+	domainG2 []byte
+}
 
 func NewBLS12381Suite() pairing.Suite {
 	return &Suite{}
 }
 
+func (s *Suite) SetDomainG1(dst []byte) {
+	s.domainG1 = dst
+}
+
 func (s *Suite) G1() kyber.Group {
-	return NewGroupG1()
+	return NewGroupG1(s.domainG1)
+}
+
+func (s *Suite) SetDomainG2(dst []byte) {
```

Comment:
Instead of `SetDomainGX`, why not creating a new suite directly with the domains ? The flow feels much better and safer:
```go
s := NewSuite()
s.SetDomainG1(myDomainG1)
s.SetDomainG2(myDomainG2)
g1 := s.G1()
g2 := s.G2()
```
vs
```go
s := NewSuiteWithDST(domainG1,domainG2)
g1 := s.G1()
g2 := s.G2()
```
it's shorter and you can't forget to set one or the other - I would go as far as even changing the API of `NewSuite(domain1,domain2)`to make sure it's being used, and in drand we set the same domain for both for the current fastnet.

Reply:
I hesitated between both and decided the path without changing our main "entry point"  was the cleanest since most of the users of this lib are using it on G2 anyway and aren't impacted by the change if I don't touch `NewSuite`, but it's true I could just have another constructor that's explicit. Let's do that.